test/fakeintake/client/client.go

```diff
@@ -96,6 +97,17 @@ func NewClient(fakeIntakeURL string) *Client {
 	}
 }
 
+// NewClient creates a new fake intake client and requires it to be healthy.
+// fakeIntakeURL: the host of the fake Datadog intake server
+func NewClient(t *testing.T, fakeIntakeURL string) *Client {
+	client := newClientNoWait(fakeIntakeURL)
+	require.EventuallyWithT(t, func(ct *assert.CollectT) {
+		assert.NoError(ct, client.GetServerHealth())
+	}, 5*time.Minute, 20*time.Second)
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
```suggestion
	}, 5*time.Minute, 500*time.Millisecond)
```
Would be more aggressive here, the task is created and its IP address ready, I would expect it to be up in less than one minute. Interval can be more aggressive, no strong opinion on reducing the timeout